src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -46,21 +44,26 @@ public void Write(ArraySegment<byte> buffer, Action<Exception, object> callback,
         public class ThisWriteReq : UvWriteReq
         {
             private static readonly Action<UvWriteReq, int, Exception, object> _writeCallback = WriteCallback;
+
             private static void WriteCallback(UvWriteReq req, int status, Exception error, object state)
             {
                 ((ThisWriteReq)state).OnWrite(req, status, error);
             }
 
             SocketOutput _self;
-            ArraySegment<byte> _buffer;
+            byte[] _buffer;
```

Comment:
The only code path into the Contextualize currently creates a copy of the incoming ArraySegment. I figured I can simplify the code down to the P/Invoke.